pkg/worker/verifier/client_helpers.go

```diff
@@ -5,33 +5,53 @@ import (
 	"errors"
 	"time"
 
+	"github.com/redpanda-data/kgo-verifier/pkg/util"
 	log "github.com/sirupsen/logrus"
 	"github.com/twmb/franz-go/pkg/kerr"
 	"github.com/twmb/franz-go/pkg/kgo"
 	"github.com/twmb/franz-go/pkg/kmsg"
 	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/kafka"
 )
 
+type OffsetResult struct {
+	offset int64
+	err    error
+}
+
 // Try to get offsets, with a retry loop in case any partitions are not
 // in a position to respond.  This is useful to avoid terminating if e.g.
 // the cluster is subject to failure injection while workload runs.
 func GetOffsets(client *kgo.Client, topic string, nPartitions int32, t int64) []int64 {
 	wait_t := 2 * time.Second
+	combinedResult := make([]int64, nPartitions)
+	haveResult := make([]bool, nPartitions)
+
+	req := formOffsetsReq(topic, nPartitions, t)
 	for {
-		result, err := getOffsetsInner(client, topic, nPartitions, t)
-		if err != nil {
-			log.Warnf("Retrying getOffsets in %v", wait_t)
-			time.Sleep(wait_t)
-		} else {
-			return result
+		result := attemptGetOffsets(client, topic, nPartitions, req)
+		var seenPartitions = int32(0)
+		for i := 0; i < int(nPartitions); i++ {
+			if result[i].err == nil {
+				if !haveResult[i] {
+					seenPartitions += 1
```

Comment:
you are incrementing seenPartitions only when "result[i] is not error" this means that this method will succeed only after you get a complete successful result which is the same behavior as before from what i read

you are also incrementing seenPartitions only when you get the result for the first time so it won't ever actually succeed if there is at least one partial failure. seenPartitions == nPartitions won't ever be true in that case